tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -142,17 +146,93 @@ private static string ConstructErrorMessage(List<IError> executionErrors)
+                    if (locations != null)
+                    {
+                        ConstructErrorLocationsMessage(builder, tab, locations);
+
+                        var joinedLocations = new List<string>();
+                        foreach (var location in (System.Collections.IEnumerable)locations)
+                        {
+                            if (location.TryDuckCast<ErrorLocationStruct>(out var locationProxy))
+                            {
+                                joinedLocations.Add($"{locationProxy.Line}:{locationProxy.Column}");
+                            }
+                        }
+
+                        eventAttributes.Add(new KeyValuePair<string, object>("locations", joinedLocations.ToArray()));
                     }
 
-                    ConstructErrorLocationsMessage(builder, tab, executionError.Locations);
                     builder.AppendLine($"{tab}}},");
+
+                    var path = executionError.Path;
+                    if (path.Name != null)
+                    {
+                        var pathAttribute = new List<string>();
+                        pathAttribute.Add(path.Name);
+                        eventAttributes.Add(new KeyValuePair<string, object>("path", pathAttribute.ToArray()));
+                    }
+
+                    var code = executionError.Code;
+                    if (code != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("code", code));
+                    }
+
+                    var exception = executionError.Exception;
+                    if (exception != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("stacktrace", exception.StackTrace));
+                    }
+
+                    var extensions = executionError.Extensions;
+                    if (extensions != null)
+                    {
+                        var configuredExtensions = Tracer.Instance.Settings.GraphQLErrorExtensions;
+
+                        var keys = extensions.Keys.ToList();
+                        for (int j = 0; j < keys.Count; j++)
+                        {
+                            if (configuredExtensions.Contains(keys[j]))
+                            {
+                                var key = keys[j];
+                                var value = extensions[key];
+
+                                if (value == null)
+                                {
+                                    value = "null";
+                                }
+                                else if (value is Array array)
+                                {
+                                    var stringArray = new string[array.Length];
+                                    for (int k = 0; k < array.Length; k++)
+                                    {
+                                        stringArray[k] = array.GetValue(k)?.ToString() ?? "null";
```

Comment:
Do you need to check the type of `array.GetValue(k)` here to ensure that it is `!(value is int || value is double || value is float || value is bool)` like below?

Reply:
Another thing worth following up on is whether arrays should be added to the attribute _as an array_ or _as a string_. If it's supposed to be returned as a string, then you can use a `StringBuilder` to build up the string

Reply:
[Based on this system-test](https://github.com/DataDog/system-tests/pull/3986/files#diff-211461fee0bb868bcd8a5fab9f61c474e10235b04c8c98767664aa1a4b52461fR76-R82), it does look like the attribute value should be a string.

So you could either build the string manually with a StringBuilder or you could try to use our vendored Newtonsoft.Json library to serialize the array value into a string and store that as the attribute value